misc/l10n_testing.md

```diff
@@ -3,55 +3,98 @@
 
+Untranslated strings appear in a user interface when developers don't include that string in the string resource files available for localization, but instead in their code. It's a practice called "hardcoding." The best way to correct this error is to make sure that Pontoon does not contain the string for translation (or that it's not in the string resource files) and then contact developers using bugzilla to make the string available for localization.
+
 ### Style bugs
 
+Mozilla's official style guide serves as a primer to help communities create their own style guides. Style errors are when the tone or appearance of a translation does not match the tone or appearance described in a style guide. These can be corrected by making sure that localizers understand the style guide rules and follow them when translating.
+
 ### Number, date, currency
 
+Every language and region displays numbers, dates, and currency differently. These errors occur when the user sees an incorrect number format, date format, or currency for the language and region of their localized Firefox. This can be fixed by filing a bug in Bugzilla.
+
 ### XML parsing errors
 
+XML parsing errors (aka "the yellow screen of death") cause Firefox to display a yellow error page. This is often due to a broken translated string in that localization of Firefox which causes the Firefox build to break. With l20n, these errors will no longer be relevant.
+
 ### Exposed variables and placeholders
 
+Variables and placeholders become exposed in the user interface when the translated string has a typo within the variable or placeholder. Because it is not exactly what the software or website expected to find for that variable, the system simply displays the variable rather than its value. A very common example of this is with the variable &browserShortName. If that variable is typed correctly in the translation, it will return the variable's value (in this case, it's "Firefox"). If it contains a typo, the user will see the variable instead (e.g., &browsershortname). These are discoverable in the Mozilla localization dashboards and can be corrected in Pontoon.
+
 ### Broken access keys
 
+Access keys break when the same key is assigned to multiple functions within the same menu. An example is if the function "Print" and the function "Save" were both assigned the access key "s." You can find broken access keys in your Firefox localization by looking at the list of access key errors in [Transvision] (https://transvision.mozfr.org/accesskeys/) for your locale.
+
 ### Broken language preferences
 
+Users experience broken language preferences in Firefox when they are left unchanged from the en-US default set of language preferences. This can lead to the user receiving web pages in a language they're unfamiliar with by default, even when the web page is localized into their native language. It can be fixed in toolkit/chrome/global/intl.properties in the accept-lang setting.
```

Comment:
need to clarify the definition of language preferences, and the fact that if only affects web content, not UI.